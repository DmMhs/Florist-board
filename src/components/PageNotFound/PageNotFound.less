.PageNotFound {
  min-height: calc(100vh - 7rem);
  border: 1px solid transparent;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  h2 {
    color: #333333;
    font-weight: normal;
    i {
      font-size: 2.5rem;
    }
  }
}
