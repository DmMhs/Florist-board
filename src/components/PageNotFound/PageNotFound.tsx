import React, { useContext } from 'react';

import './PageNotFound.less';
import { AppContext } from '../../AppContext';
import labels from '../../config/labels';

const pageNotFound = () => {
  const context = useContext(AppContext);
  return (
    <div className="PageNotFound">
      <h2>
        <span className="message">
          {context.state.lang === 'en'
            ? labels.en.pages.pageNotFound
            : labels.uk.pages.pageNotFound}
        </span>{' '}
        <i className="far fa-grimace" />
      </h2>
    </div>
  );
};

export default pageNotFound;
