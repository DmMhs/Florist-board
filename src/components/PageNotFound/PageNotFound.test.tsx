import React from 'react';
import { shallow, mount } from 'enzyme';

import PageNotFound from './PageNotFound';
import AppContextProvider from '../../AppContext';
import { BrowserRouter } from 'react-router-dom';

describe('PageNotFound works as expected', () => {
  it('PageNotFound component matches a snapshot', () => {
    const wrapper = shallow(<PageNotFound />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Message is not empty', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <PageNotFound />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();

    context.setState({
      lang: 'en'
    });

    expect(wrapper.find('.message').text().length).toBeGreaterThan(0);
  });
});
