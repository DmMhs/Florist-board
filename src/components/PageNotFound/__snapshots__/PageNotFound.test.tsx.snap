// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component matches a snapshot 1`] = `
<div
  className="PageNotFound"
>
  <h2>
    <span
      className="message"
    >
      Сторінку не знайдено
    </span>
     
    <i
      className="far fa-grimace"
    />
  </h2>
</div>
`;
